src/modular/boxed_monty_form.rs

```diff
@@ -244,8 +244,8 @@ impl Monty for BoxedMontyForm {
         BoxedMontyParams::new_vartime(modulus)
     }
 
-    fn new(value: Self::Integer, params: Self::Params) -> Self {
-        BoxedMontyForm::new(value, params)
+    fn new(value: &Self::Integer, params: Self::Params) -> Self {
+        BoxedMontyForm::new(value.clone(), params)
```

Comment:
Hmm, I know I already hit approve but this has me second guessing if it's worth it. `MontyForm` is a `Copy` type but `BoxedMontyForm` is a heap-backed type and this is adding another heap allocation which could otherwise be avoided.

Reply:
Yeah my heart sank a bit too when I had to do this. Then I figured that 1) a Box is cheap to clone and 2) the unboxed variant is probably what users that care about performance would prefer, so perhaps it makes sense to accommodate them?

Reply:
The heap allocation is a tangible performance hit, whereas passing a `Copy` type by reference may have no performance impact, particularly with inlining

Reply:
As a user of the API before this PR I find that I end up having to clone my `Integer`s more often than I'd like, given that `MontyForm` just reads the data. But people instantiating with `BoxedMontyForm` will need the same clones in their code yeah? It's less annoying for them I guess, because if they read the code they see that the `value` is actually mutated in place.

At the end of the day I think it comes down to what the API is trying to communicate. Taking ownership of the arguments signals that the users aren't supposed to re-use the values, but when it comes to "number types" I think the expectation is that you can actually continue using it (which is the case for `Copy` types of course).

Maybe this is tricky to decide because the underlying impls are so different. Would it make sense to try to refactor `BoxedMontyForm::new` to take a ref and then do the minimum requried amount of copying?

Reply:
Why are you cloning a `Copy` type?

Reply:
Because I'm using it in generic code, through the `Integer` trait (and the `Monty` associated type`) so I don't know if my users are going to use it with `MontyForm` or `BoxedMontyForm`.

Reply:
Aah yeah, that's a good point